Servers/UI/OJS.Servers.Ui/ClientApp/src/common/urls/compose-client-urls.ts

```diff
@@ -0,0 +1,35 @@
+import { CONTESTS_PATH } from './client-urls';
+
+const composeParticipationTypeResultsFullRoute =
+    (id: number, participationType: string, isSimple: boolean) => `/${CONTESTS_PATH}/${id}/${participationType}/results/${isSimple
+        ? 'simple'
+        : 'full'}`;
+
+const composeContestsWithSelectedCategoryAndStrategyUrl =
+    (categoryId?: number, strategyId?: number) => {
+        const returnUrl = `/${CONTESTS_PATH}`;
+        const prefixes = [ `${categoryId
+            ? `category=${categoryId}`
+            : ''}`, `${strategyId
+            ? `strategy=${strategyId}`
+            : ''}` ];
+        const composedString = prefixes.filter((prefix) => prefix).join('&');
+        if (composedString.length > 0) {
+            return `${returnUrl}?${composedString}`;
+        }
+
+        return returnUrl;
+    };
+
+const composeContestParticipateUrl = (isCompete: boolean, id: number) => {
+    if (isCompete) {
+        return `/${CONTESTS_PATH}/${id}/compete`;
+    }
+    return `/${CONTESTS_PATH}/${id}/practice`;
+};
+
+export {
+    composeParticipationTypeResultsFullRoute,
+    composeContestsWithSelectedCategoryAndStrategyUrl,
```

Comment:
Let's make a good convention from now on, as this file will be filled with urls.

Will we use `Route` or `Url` at the end?
`get` or `compose` at the beginning - I like get, as it's shorter.
I favor shorter names, but descriptive enough for the purpose of the url, not so much about the parameters.
- `getContestResultsUrl` - You should use it when you want contest results (full or simple does not matter, also participation type does not matter - it's all in here, in the parameters)
- `getAllContestsUrl` - Use when you want all contests with or without filters (depending on the params).

Reply:
sounds good enough for me, just pushed changes accordingly